src/CoreMedia/CMSampleBuffer.cs

```diff
@@ -834,6 +841,72 @@ public CMSampleBufferError CopyPCMDataIntoAudioBufferList (int frameOffset, int
 
 			return new CMSampleBuffer (buffer, true);
 		}
+
+		[iOS (12,2)]
+		[TV (12,2)]
+		[Mac (10,14,4, onlyOn64: true)]
+		[DllImport (Constants.CoreMediaLibrary)]
+		static unsafe extern /* OSStatus */ CMSampleBufferError CMSampleBufferCreateForImageBufferWithMakeDataReadyHandler (
+			/* CFAllocatorRef CM_NULLABLE */ IntPtr allocator,
+			/* CVImageBufferRef CM_NONNULL */ IntPtr imageBuffer,
+			/* Boolean */ [MarshalAs (UnmanagedType.I1)] bool dataReady,
+			/* CMSampleBufferMakeDataReadyHandler CM_NULLABLE */ IntPtr makeDataReadyHandler,
+			/* CMVideoFormatDescriptionRef CM_NONNULL */ IntPtr formatDescription,
+			/* const CMSampleTimingInfo* CM_NONNULL */ CMSampleTimingInfo [] sampleTiming,
+			/* CM_RETURNS_RETAINED_PARAMETER CMSampleBufferRef CM_NULLABLE * CM_NONNULL */ out IntPtr sampleBufferOut);
+
+		internal delegate CMSampleBufferError DCMSampleBufferMakeDataReadyHandler (IntPtr block, IntPtr sbuf);
+
+		static internal class DCMSampleBufferMakeDataReadyHandlerTrampoline {
+			static internal readonly DCMSampleBufferMakeDataReadyHandler Handler = Invoke;
+
+			[MonoPInvokeCallback (typeof (DCMSampleBufferMakeDataReadyHandler))]
+			static unsafe CMSampleBufferError Invoke (IntPtr block, IntPtr sbuf)
+			{
+				var descriptor = (BlockLiteral *) block;
+				var del = (CMSampleBufferMakeDataReadyHandler) descriptor->Target;
+				using (var b = new CMSampleBuffer (sbuf))
+					return del (b);
+			}
+		}
+
+		[iOS (12,2)]
+		[TV (12,2)]
+		[Mac (10,14,4, onlyOn64: true)]
+		[BindingImpl (BindingImplOptions.Optimizable)]
+		public static CMSampleBuffer CreateReadyWithImageBuffer (CVImageBuffer imageBuffer,
+			bool dataReady, CMSampleBufferMakeDataReadyHandler makeDataReadyHandler,
+			CMFormatDescription formatDescription, CMSampleTimingInfo [] sampleTiming, out CMSampleBufferError error)
```

Comment:
`sampleTiming` must contain (at least) one element, so that should be validated. In fact, the input should probably be just `CMSampleTimingInfo`, not an array of them.

Reply:
I think it's an array - but I've read too many variations... I'll double check :)
good point on empty - won't be better than `null`